index.js

```diff
@@ -141,44 +186,68 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+// *** ANSWER ***
+let num = 0
+
+function takeATicketNumber(line) {
+  num++
+  line.push(num)
+
+  return `Welcome. You are ticket number ${num}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log("%c----------", "color: red") 
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+
+function ticketNumberGeneratorFunc() {
+  let num = 0;
+  function takeATicketNumber(arr){
+    num++;
+    arr.push;
+
+    return `Welcome. You are ticket number ${num}`
+  }
+
+  return takeATicketNumber
+}
```

Comment:
This is great. we can refactor by removing line 235 and adding the return in front of line 228. We can also make the function on 228 an anonymous function if you want.